akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSlidingSpec.scala

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.actor.ActorSystem
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.stream.{ ActorMaterializer, ActorMaterializerSettings }
+import akka.stream.testkit._
+import com.typesafe.config.ConfigFactory
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class FlowSlidingSpec extends AkkaSpec {
+  import system.dispatcher
+  val settings = ActorMaterializerSettings(system)
+    .withInputBuffer(initialSize = 2, maxSize = 16)
+
+  implicit val mat = ActorMaterializer(settings)
+
+  "Sliding" must {
+    "work with n = 3, step = 1" in assertAllStagesStopped {
+      Source(1 to 6).sliding(n = 3, step = 1).runForeach(testActor ! _) onSuccess {
+        case _ ⇒ testActor ! "done"
+      }
+      expectMsg(Vector(1, 2, 3))
+      expectMsg(Vector(2, 3, 4))
+      expectMsg(Vector(3, 4, 5))
+      expectMsg(Vector(4, 5, 6))
+      //expectMsg(Vector.empty[Int])
+      expectMsg("done")
+    }
+
+    "work with n = 3, step = 2" in assertAllStagesStopped {
+      Source(1 to 6).sliding(n = 3, step = 2).runForeach(testActor ! _) onSuccess {
+        case _ ⇒ testActor ! "done"
+      }
+      expectMsg(Vector(1, 2, 3))
+      expectMsg(Vector(3, 4, 5))
+      expectMsg(Vector(5, 6))
+      //expectMsg(Vector.empty[Int])
+      expectMsg("done")
+    }
+
+    "work with n = 3, step = 3" in assertAllStagesStopped {
+      Source(1 to 6).sliding(n = 3, step = 3).runForeach(testActor ! _) onSuccess {
+        case _ ⇒ testActor ! "done"
+      }
+      expectMsg(Vector(1, 2, 3))
+      expectMsg(Vector(4, 5, 6))
+      //expectMsg(Vector.empty[Int])
+      expectMsg("done")
+    }
+
+    "work with n = 2, step = 3" in assertAllStagesStopped {
+      Source(1 to 6).sliding(n = 2, step = 3).runForeach(testActor ! _) onSuccess {
+        case _ ⇒ testActor ! "done"
+      }
+      expectMsg(Vector(1, 2))
+      expectMsg(Vector(4, 5))
+      //expectMsg(Vector.empty[Int])
+      expectMsg("done")
+    }
+
+    "work with n = 2, step = 1" in assertAllStagesStopped {
+      Source(1 to 6).sliding(n = 2, step = 1).runForeach(testActor ! _) onSuccess {
+        case _ ⇒ testActor ! "done"
+      }
+      expectMsg(Vector(1, 2))
+      expectMsg(Vector(2, 3))
+      expectMsg(Vector(3, 4))
+      expectMsg(Vector(4, 5))
+      expectMsg(Vector(5, 6))
+      //expectMsg(Vector.empty[Int])
+      expectMsg("done")
+    }
```

Comment:
can you add a test that `Source.empty.sliding(1)` completes right away (without accidentally yielding a `Vector.empty`)